drgn_tools/runq.py

```diff
@@ -65,20 +67,31 @@ def _print_cfs_runq(runqueue: Object) -> None:
         print("     [no tasks queued]")
 
 
-def run_queue(prog: Program) -> None:
+def run_queue(prog: Program, min_run_time_seconds: int = 0) -> None:
     """
-    Print tasks which are in the RT and CFS runqueues on each CPU
+    Print tasks which are in the RT and CFS runqueues on each CPU. Specify min_run_time_seconds to x to only print
+    processes running more than x seconds.
+
+    :param prog: drgn program
+    :param min_run_time_seconds: int
     """
 
     # _cpu = drgn.helpers.linux.cpumask.for_each_online_cpu(prog)
     for cpus in for_each_online_cpu(prog):
         runqueue = per_cpu(prog["runqueues"], cpus)
-        comm = escape_ascii_string(runqueue.curr.comm.string_())
-        pid = runqueue.curr.pid.value_()
+        curr_task_addr = runqueue.curr.value_()
+        curr_task = runqueue.curr[0]
```

Comment:
Why use `runqueue.curr[0]`, `runqueue.curr` not working?

Reply:
This change is due to a bug we observed . The pointer object 'runqueue.curr' can change in between when we are getting its attributes, so we need to dereference it first to get the value object for safety.

Reply:
i see